### PR TITLE
Extend recommended readings section

### PR DESCRIPTION
**Pull request by frankharkins**
For the new "basics of quantum information course", there's a need to provide some extra information on these resources (e.g. what they are, and why we recommend them).

## Changes

This PR extends the "external recommended readings" section of the course overview page to:
1. Accept an optional `author` value for each link
2. Accept an optional `description` value for each link
3. Accept an optional `preamble` value

- If `author` is provided, the name of the link becomes italic,
- if `description` is provided, this text appears after the link.
- If `preamble` is provided, this text appears at the top of the section.
- If no field is provided, it should behave exactly as before.

## How to read this PR

- b920b86915370ee940dc6bd6f8db18582c4322ee implements the code changes.
- 796f42ca57a0dbfcccc5c225455ae1362fd87d4d & 6225dae1e2eb5c44af153768351e8860f3c46780 reformat the existing values to use the `author` field.
- 5be0cc23466198f4d0440cc6c5e23805bdc84473 adds the `preamble` feature.

## Screenshots

No external recommended readings currently have a 'description' or 'preamble', so here's an example from the basics course:

Screenshot 2022-10-25 at 13 09 28

### CODE REVIEW

File: components/learn/ExternalRecommendedReadings.vue [hunk 1]
```diff
@@ -21,8 +21,15 @@
...
         >
-          {{ link.label }}
+          <span v-if="link.author">
+            {{ link.author }}
+          </span>
+          <span :class="link.author ? 'external-recommended-readings__title' : ''">
+            {{ link.label }}
+          </span>
         </AppLink>
+        {{ link.description ? '-' : '' }}
```
Comment (techtolentino) [inline components/learn/ExternalRecommendedReadings.vue:31]: For these two lines, you should be able to refactor into something like so:

```html
<span v-if="link.description"> - {{ link.description }}</span>
```

File: components/learn/ExternalRecommendedReadings.vue [hunk 2]
```diff
@@ -49,12 +61,19 @@ export default class ExternalRecommendedReadings extends Vue {
...
   }
 
+  li {
+    margin-bottom: $spacing-02;
+  }
+
   &__link {
-    display: block;
-    margin-bottom: $spacing-01;
+    /* display: block; */
```
Comment (techtolentino) [inline components/learn/ExternalRecommendedReadings.vue:69]: this can be removed

File: components/learn/ExternalRecommendedReadings.vue [hunk 3]
```diff
@@ -33,6 +40,11 @@ import Vue from 'vue'
...
 import { GeneralLink } from '~/constants/appLinks'
 
+export type RecommendedReading = GeneralLink & {
+    description?: string;
+    author?: string;
+    }
+
 @Component
 export default class ExternalRecommendedReadings extends Vue {
   @Prop({ type: Array, required: true }) links!: GeneralLink[]
```
Comment (techtolentino) [inline components/learn/ExternalRecommendedReadings.vue:50]: Should this, then be:

```js
@Prop({ type: Array, required: true }) links!: RecommendedReading[]
```

?

File: pages/learn/course/introduction-course.vue [hunk 4]
```diff
@@ -44,18 +45,20 @@ export default class IntroductionCoursePage extends CourseOverviewPage {
   }
 
   references: string[] = []
-  links: GeneralLink[] = [
+  links: RecommendedReading[] = [
     {
       url: 'https://math.mit.edu/~gs/linearalgebra/',
-      label: 'Gilbert Strang, Introduction to Linear Algebra, 2016',
+      author: 'Gilbert Strang,',
+      label: 'Introduction to Linear Algebra.',
```
Comment (techtolentino) [inline pages/learn/course/introduction-course.vue:52]: remove `.`

File: pages/learn/course/introduction-course.vue [hunk 5]
```diff
@@ -44,18 +45,20 @@ export default class IntroductionCoursePage extends CourseOverviewPage {
...
       segment: {
         cta: 'introduction-to-linear-algebra',
         location: 'external-recommended-readings'
       }
     },
     {
       url: 'https://machinelearningmastery.com/gentle-introduction-linear-algebra/',
-      label: 'Jason Brownlee, A Gentle Introduction to Linear Algebra, 2018',
+      author: 'Jason Brownlee,',
+      label: 'A Gentle Introduction to Linear Algebra.',
```
Comment (techtolentino) [inline pages/learn/course/introduction-course.vue:61]: remove `.`

**Review by techtolentino — CHANGES_REQUESTED** [review]
Looking good @frankharkins 

Left a few minor comments.

Thanks!

File: components/learn/ExternalRecommendedReadings.vue (hunk 1 above)
Comment (frankharkins) [reply components/learn/ExternalRecommendedReadings.vue]: Cool, thanks

File: components/learn/ExternalRecommendedReadings.vue (hunk 3 above)
Comment (frankharkins) [reply components/learn/ExternalRecommendedReadings.vue]: Whoops, yes - thanks

**Review by vabarbosa — APPROVED** [review]
thank you @frankharkins!!

**Pull request merged**